File: fixtures/elixir/test/calculator_test.exs
defmodule CalculatorTest do
  use ExUnit.Case, async: true

  alias Calculator.DivisionByZeroError

  test "adds two numbers" do
    calc = Calculator.new()
    {result, _calc} = Calculator.add(calc, 2, 3)
    assert result == 5
  end

  test "subtracts two numbers" do
    calc = Calculator.new()
    {result, _calc} = Calculator.subtract(calc, 5, 3)
    assert result == 2
  end

  test "multiplies two numbers" do
    calc = Calculator.new()
    {result, _calc} = Calculator.multiply(calc, 4, 3)
    assert result == 12
  end

  test "divides two numbers" do
    calc = Calculator.new()
    {result, _calc} = Calculator.divide(calc, 10, 2)
    assert result == 5.0
  end

  test "raises on division by zero" do
    calc = Calculator.new()

    assert_raise DivisionByZeroError, fn ->
      Calculator.divide(calc, 1, 0)
    end
  end

  test "tracks history" do
    calc = Calculator.new()
    {_r1, calc} = Calculator.add(calc, 1, 2)
    {_r2, calc} = Calculator.multiply(calc, 3, 4)
    assert Calculator.get_history(calc) == [3, 12]
  end

  test "clears history" do
    calc = Calculator.new()
    {_r1, calc} = Calculator.add(calc, 1, 2)
    calc = Calculator.clear_history(calc)
    assert Calculator.get_history(calc) == []
  end
end

File: fixtures/javascript/test/calculator.test.js
import { describe, it } from 'node:test';
import assert from 'node:assert/strict';
import { Calculator } from '../src/calculator.js';
import { DivisionByZeroError } from '../src/errors.js';

describe('Calculator', () => {
  it('adds two numbers', () => {
    const calc = new Calculator();
    assert.strictEqual(calc.add(2, 3), 5);
  });

  it('subtracts two numbers', () => {
    const calc = new Calculator();
    assert.strictEqual(calc.subtract(5, 3), 2);
  });

  it('multiplies two numbers', () => {
    const calc = new Calculator();
    assert.strictEqual(calc.multiply(4, 3), 12);
  });

  it('divides two numbers', () => {
    const calc = new Calculator();
    assert.strictEqual(calc.divide(10, 2), 5);
  });

  it('throws on division by zero', () => {
    const calc = new Calculator();
    assert.throws(() => calc.divide(1, 0), DivisionByZeroError);
  });

  it('tracks history', () => {
    const calc = new Calculator();
    calc.add(1, 2);
    calc.multiply(3, 4);
    assert.deepStrictEqual(calc.getHistory(), [3, 12]);
  });

  it('clears history', () => {
    const calc = new Calculator();
    calc.add(1, 2);
    calc.clearHistory();
    assert.deepStrictEqual(calc.getHistory(), []);
  });
});
